t/api/10-jobgroups.t

```diff
@@ -178,47 +228,117 @@ subtest 'update job group' => sub() {
-                default_priority               => 50,
-                carry_over_bugrefs             => 1,
+                sort_order                     => 123,
+                keep_logs_in_days              => 22,                # changed through inheritance
+                keep_important_logs_in_days    => 44,                # changed through inheritance
+                default_priority               => 70,
+                carry_over_bugrefs             => 0,
                 description                    => 'Test',
                 template                       => undef,
-                keep_results_in_days           => 365,
-                keep_important_results_in_days => 366,          # changed through inheritance
+                keep_results_in_days           => 222,               # changed through inheritance
+                keep_important_results_in_days => 333,               # changed through inheritance
                 size_limit_gb                  => 101,
                 build_version_sort             => 0,
-                id                             => 1001,
+                id                             => $opensuse_group,
                 exclusively_kept_asset_size    => undef,
             },
         ],
         'list updated job group'
     );
+
+    $t->put_ok(
+        "/api/v1/job_groups/$opensuse_group",
+        form => {
+            name                           => 'opensuse',
+            keep_logs_in_days              => 20,
+            keep_important_logs_in_days    => 40,
+            keep_results_in_days           => 200,
+            keep_important_results_in_days => 300,
+        })->status_is(200, 'defaults overridden');
+
+    $t->get_ok("/api/v1/job_groups/$opensuse_group")->status_is(200);
+    is_deeply(
+        $t->tx->res->json,
+        [
+            {
+                name                           => 'opensuse',
+                parent_id                      => $new_id,
+                sort_order                     => 123,
+                keep_logs_in_days              => 20,                # inherited value overridden
+                keep_important_logs_in_days    => 40,                # inherited value overridden
+                default_priority               => 70,
+                carry_over_bugrefs             => 0,
+                description                    => 'Test',
+                template                       => undef,
+                keep_results_in_days           => 200,               # inherited value overridden
+                keep_important_results_in_days => 300,               # inherited value overridden
+                size_limit_gb                  => 101,
+                build_version_sort             => 0,
+                id                             => $opensuse_group,
+                exclusively_kept_asset_size    => undef,
+            },
+        ],
+        'list updated job group with overridden defaults'
+    );
```

Comment:
Not sure if big blocks like this area easily readable and maintainable. How about just checking for some of the values? E.g.

```perl
    my $ret = $t->tx->res->json->[0];
    is $ret->{name}, 'opensuse', 'correct job group is updated';
    is $ret->{keep_important_logs_in_days}, 40, 'inherited keep_important_logs_in_days overridden';

```

Reply:
You´re right, for these ones we don´t need to check all the values!